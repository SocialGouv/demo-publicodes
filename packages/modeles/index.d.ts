// Currenty we systematically bundle all the rules even if we only need a
// sub-section of them. We might support "code-splitting" the rules in the
// future.
import { Rule } from "publicodes";

export type { Names as OrientationCovidNames } from "./dist/orientation-covid/index.d.ts";

//export type DottedName = Names;

declare let rules: Record<key, Rule>;

declare let models: Record<key, rules>;

export default models;
