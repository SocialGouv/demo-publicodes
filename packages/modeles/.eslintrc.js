module.exports = {
  extends: ["prettier", "eslint:recommended"],
};
