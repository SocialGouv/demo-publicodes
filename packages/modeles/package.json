{
  "name": "@socialgouv/publicodes-demo-modeles",
  "version": "1.0.0",
  "description": "Des règles publicodes de démo",
  "type": "module",
  "main": "./dist/index.js",
  "module": "dist/index.js",
  "types": "./index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/socialgouv/demo-publicodes.git",
    "directory": "packages/modeles"
  },
  "homepage": "https://github.com/socialgouv/demo-publicodes",
  "license": "MIT",
  "files": [
    "dist",
    "index.d.ts"
  ],
  "devDependencies": {
    "eslint-config-prettier": "^8.5.0",
    "jest": "^29.3.1",
    "js-yaml": "^4.1.0",
    "onchange": "^7.1.0",
    "publicodes": "^1.0.0-beta.55"
  },
  "peerDependencies": {
    "publicodes": "^1.0.0-beta.55"
  },
  "scripts": {
    "build": "node ./scripts/build-rules.js",
    "start": "yarn run build && onchange 'règles/**/*.yaml' -- yarn run build",
    "clean": "rimraf dist node_modules",
    "prepack": "yarn run build",
    "test": "yarn node --experimental-vm-modules $(yarn bin jest)"
  },
  "dependencies": {
    "camelcase": "^7.0.0"
  }
}
