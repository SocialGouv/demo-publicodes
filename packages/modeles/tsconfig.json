{
	"extends": "tsconfig/base.json",
	"compilerOptions": {
		"allowJs": true,
		"noEmit": true,
		"strict": true
	},
	"include": ["index.d.ts", "dist/**/*.ts"]
}
