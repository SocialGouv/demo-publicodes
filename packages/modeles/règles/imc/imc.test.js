/* eslint-env node18 */
import yaml from "js-yaml";
import Engine, { formatValue } from "publicodes";

import modeles from "@socialgouv/publicodes-demo-modeles";
const rules = yaml.load(modeles.imc.yaml);

const tests = [
  {
    poids: 78,
    taille: 178,
  },
  {
    poids: 58,
    taille: 178,
  },
  {
    poids: 92,
    taille: 178,
  },
  {
    poids: 92,
    taille: 158,
  },
  {
    poids: 117,
    taille: 158,
  },
  {
    poids: 45,
    taille: 158,
  },
  {
    poids: 32,
    taille: 158,
  },
];

test.each(tests)(".imc($poids, $taille)", ({ poids, taille }) => {
  const engine = new Engine(rules);
  engine.setSituation({
    poids: poids,
    taille: taille,
  });
  const evaluated = engine.evaluate("résultat");
  expect(evaluated.nodeValue).toMatchSnapshot();
});

test("doit renvoyer le bon IMC pour T=170,P=80)", () => {
  const engine = new Engine(rules);
  engine.setSituation({
    poids: "80kg",
    taille: "170cm",
  });
  const evaluated = engine.evaluate("imc");
  // teste la valeur brute
  expect(evaluated.nodeValue).toEqual(27.68166089965398);
  // teste la valeur formatée
  expect(formatValue(evaluated)).toEqual("27,68\u00A0kg / m²");
});

test("doit renvoyer la bonne interpretation pour T=180,P=80)", () => {
  const engine = new Engine(rules);
  engine.setSituation({
    poids: "80kg",
    taille: "180cm",
  });
  const evaluated = engine.evaluate("résultat");
  expect(evaluated.nodeValue.trim())
    .toEqual(`Votre IMC est de 24,69\u00A0kg / m²
Interprétation: poids normal`);
});

test("doit renvoyer la bonne interpretation pour T=150,P=90)", () => {
  const engine = new Engine(rules);
  engine.setSituation({
    poids: "90kg",
    taille: "150cm",
  });
  const evaluated = engine.evaluate("résultat");
  expect(evaluated.nodeValue.trim()).toEqual(
    `Votre IMC est de 40\u00A0kg / m²\nInterprétation: obésité morbide ou massive`
  );
});
