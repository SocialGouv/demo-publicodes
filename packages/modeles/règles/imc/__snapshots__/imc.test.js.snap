// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.imcu(32, 158) 1`] = `
"Votre IMC est de 12,82 kg / m²
Interprétation: dénutrition
"
`;

exports[`.imcu(45, 158) 1`] = `
"Votre IMC est de 18,03 kg / m²
Interprétation: maigreur
"
`;

exports[`.imcu(58, 178) 1`] = `
"Votre IMC est de 18,31 kg / m²
Interprétation: maigreur
"
`;

exports[`.imcu(78, 178) 1`] = `
"Votre IMC est de 24,62 kg / m²
Interprétation: poids normal
"
`;

exports[`.imcu(92, 158) 1`] = `
"Votre IMC est de 36,85 kg / m²
Interprétation: obésité sévère
"
`;

exports[`.imcu(92, 178) 1`] = `
"Votre IMC est de 29,04 kg / m²
Interprétation: surpoids
"
`;

exports[`.imcu(117, 158) 1`] = `
"Votre IMC est de 46,87 kg / m²
Interprétation: obésité morbide ou massive
"
`;
